src/NUnitFramework/framework/Interfaces/ITestResult.cs

```diff
@@ -157,7 +157,7 @@ bool HasChildren
         /// <summary>
         /// Gets the the collection of child results.
         /// </summary>
-        System.Collections.Generic.IList<ITestResult> Children
+        System.Collections.Concurrent.ConcurrentQueue<ITestResult> Children
```

Comment:
I don't like allowing an implementation detail to leak out through the interface. We should examine how children is used. Since we can no longer use IList, we should look at how Children is used and pick another System interface if possible. If not, perhaps we should create an ITestResultChildren interface and implement it differentially according to the platform and use.